File: 백준/Gold/12865. 평범한 배낭/평범한 배낭.py
N, K = map(int, input().split(' '))

dp = [0] * (K+1)
val_list = [0]

for i in range(N):

    weight, value = map(int, input().split(' '))
    
    for j in range(K,weight - 1,-1):  # j>=weight
        # 1. input weight를 넣을 수 있으면        
        dp[j] = max(dp[j], dp[j-weight] + value)


print(dp[-1])

File: 백준/Gold/12865. 평범한 배낭/평범한 배낭.cc
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N, K;
    cin >> N >> K;

    vector<int> dp(K+1, 0);

    for (int i=0; i<N; i++) {
        int w, v;
        cin >> w >> v;
        for (int j=K; j>=w; j--) {
            dp[j] = max(dp[j], dp[j-w] + v);
        }
    }

    cout << dp[K];
}
